Drop the use of the `hosts` addon in Travis

foolip opened the pull request:
The hosts list is out of date and this isn't needed, as
tools/ci/lib.sh does the hosts fixup when needed.
    
Drive-by: remove a redundant `sudo: required`.

foolip committed:
Drop the use of the `hosts` addon in Travis

The hosts list is out of date and this isn't needed, as
tools/ci/lib.sh does the hosts fixup when needed.

Drive-by: remove a redundant `sudo: required`.

gsnedders commented on .travis.yml:
```diff
@@ -64,7 +56,6 @@ matrix:
     - name: "stability (Chrome Dev)"
       if: type = pull_request
       os: linux
-      sudo: required
```
Surely this requires sudo to write `/etc/hosts`?

gsnedders replied:
But Chrome needs `/etc/hosts` which `hosts_fixup` will try to write to, no? What description says it's redundant? (I see it manages to write it, so I don't know what description you're referring to here.)

foolip replied:
I mean the description of this PR. That doesn't say *why* it's redundant, I left that implied: because it's already at the top-level scope of `.travis.yml`.

foolip commented:
From https://travis-ci.org/web-platform-tests/wpt/builds/438685080 it looks like serve/test_functional.py inadvertently depend on /etc/hosts :)

gsnedders commented:
@jugglinmike this is your test, from 1f17212bcf6/#12557. Let me try and fix this…

gsnedders committed:
Avoid serve/test_functional.py#test_subprocess_exit relying on hosts

Without bind_address: False we use the server address for the stash,
and this obviously doesn't work without the hosts being set up.

gsnedders commented:
Passing now, yay!

gsnedders commented:
Would rather someone else reviewed this given my commit here. Also note it needs rebased to remove the REVERTME commit.
